test/Tester/CancellationTests/CancellationTokenPassingTests.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                var grains = await GetGrains<bool>();
+                var grain = grains.Item1;
+                var target = grains.Item2;
+                var tcs = new GrainCancellationTokenSource();
+                var wait = grain.CallOtherLongRunningTask(target, tcs.Token, TimeSpan.FromSeconds(10));
+                await Task.Delay(delay);
+                await tcs.Cancel();
+                await Assert.ThrowsAsync<TaskCanceledException>(() => wait);
+            }
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task InterSiloGrainCancellation()
+        {
+            await GrainGrainCancellation(true);
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task InSiloGrainCancellation()
+        {
+            await GrainGrainCancellation(false);
+        }
+
+        private async Task GrainGrainCancellation(bool interSilo)
+        {
+            foreach (var delay in cancellationDelaysInMS)
+            {
+                var grains = await GetGrains<bool>(interSilo);
+                var grain = grains.Item1;
+                var target = grains.Item2;
+                var wait = grain.CallOtherLongRunningTaskWithLocalToken(target, TimeSpan.FromSeconds(10),
+                    TimeSpan.FromMilliseconds(delay));
+                await Assert.ThrowsAsync<TaskCanceledException>(() => wait);
+            }
+        }
+
+        private async Task<Tuple<ILongRunningTaskGrain<T1>, ILongRunningTaskGrain<T1>>> GetGrains<T1>(bool placeOnDifferentSilos = true)
+        {
+            var grain = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<T1>>(Guid.NewGuid());
+            var instanceId = await grain.GetRuntimeInstanceId();
+            var target = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<T1>>(Guid.NewGuid());
+            var targetInstanceId = await target.GetRuntimeInstanceId();
+            var retriesCount = 0;
+            var retriesLimit = 7;
+            if (placeOnDifferentSilos)
+            {
+                while (instanceId.Equals(targetInstanceId))
+                {
+                    if(retriesCount >= retriesLimit) throw new Exception("Could not place grains on different silos");
+                    target = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<T1>>(Guid.NewGuid());
+                    targetInstanceId = await target.GetRuntimeInstanceId();
+                    retriesCount++;
+                }
+            }
+            else
+            {
+                while (!instanceId.Equals(targetInstanceId))
+                {
+                    if (retriesCount >= retriesLimit) throw new Exception("Could not place grains on same silo");
+                    target = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<T1>>(Guid.NewGuid());
+                    targetInstanceId = await target.GetRuntimeInstanceId();
+                    retriesCount++;
+                }
+            }
```

Comment:
nit picking: I would have 1 while loop and just have the condition be smarter in the case that `placeOnDifferentSilos` is true or false, instead of duplicating a bunch of code.